vim-patch:8.1.{820,1184,1379}

janlazo opened the pull request:
`USE_INPUT_BUF` is not defined in Neovim so I assume that the `L` flag for feedkeys() cannot be supported and skipped a test.

janlazo committed:
vim-patch:8.1.0820: test for sending large data over channel sometimes fails

Problem:    Test for sending large data over channel sometimes fails.
Solution:   Handle that the job may have finished early.  Also fix that file
            changed test doesn't work in the GUI and reduce flakyness. (Ozaki
            Kiichi, closes vim/vim#3861)
https://github.com/vim/vim/commit/e295609be20c1b065d6b489214d70e171ae49234

Cherry-pick the following from patch v8.1.0815:
- 'L' flag for feedkeys()
- test_autocmd.vim
- test_filechanged.vim

Patch v8.1.0815 cannot be ported
because it includes documentation for the Blob feature.
Patches for Blob feature are not merged yet.

Include patch v8.1.0851 changes to fix Test_getchar_zero().

janlazo committed:
vim-patch:8.1.1184: undo file left behind after running test

Problem:    Undo file left behind after running test.
Solution:   Delete the undo file. (Dominique Pelle, closes vim/vim#4279)
https://github.com/vim/vim/commit/137c14bb4f18198ed38659dcfbdfd749115c7ab5

janlazo committed:
vim-patch:8.1.1379: filechanged test hangs

Problem:    Filechanged test hangs.
Solution:   Do not check 'autoread'.
https://github.com/vim/vim/commit/8239c62067c8d40720560496b25a82662126f2a2

bfredl commented:
Hmm maybe nvim_input() works for low level input.

janlazo commented:
Tried replacing the error message with `nvim_input` so that `L` flag does not throw an error but `Test_file_changed_dialog` fails.

janlazo commented on src/nvim/api/vim.c:
```diff
@@ -251,10 +253,14 @@ void nvim_feedkeys(String keys, String mode, Boolean escape_csi)
-  if (vgetc_busy) {
-    typebuf_was_filled = true;
+  if (lowlevel) {
+    (void)nvim_input(keys);
```
It works for `Test_getchar_zero` but not `Test_file_changed_dialog()`.

bfredl commented:
what is the input used in `Test_file_changed_dialog()`? one footgun `nvim_input()` has it, that it is unreliable if the length approaches the fixed input buffer size (which ideally could become more dynamic).

janlazo commented on src/nvim/testdir/test_filechanged.vim:
```diff
@@ -0,0 +1,149 @@
+  write
+  " Need to wait until the timestamp would change by at least a second.
+  sleep 2
+  silent !echo 'extra line' >>Xchanged_r
```
I guess that the changes to `Xchanged_r` are not reflected in the buffer after `:checktime` so the assertions consistently fail.

zeertzjq replied:
This failure is actually because there is no UI active, so `do_dialog()` just chooses the default option.

janlazo committed:
vim-patch:8.1.0447: GUI scrollbar test fails with Athena and Motif

Problem:    GUI scrollbar test fails with Athena and Motif.
Solution:   When not using on-the-fly scrolling call normal_cmd().
https://github.com/vim/vim/commit/586c70cdfede55a166e3564f1cb68a299d81987d

janlazo committed:
vim-patch:8.1.1136: decoding of mouse click escape sequence is not tested

Problem:    Decoding of mouse click escape sequence is not tested.
Solution:   Add a test for xterm and SGR using low-level input.  Make
            low-level input execution with feedkeys() work.
https://github.com/vim/vim/commit/905dd905debfde403b2a18178ccc1f8e118f4f2b

justinmk commented:
https://github.com/neovim/neovim/commit/a500c5f808ccf0b678c935f00e0af4503a5bd724
